subsys/net/l2/canbus/canbus.c

```diff
@@ -200,10 +200,28 @@ static bool dest_is_translator(struct net_pkt *pkt)
 		net_pkt_lladdr_dst(pkt)->len == sizeof(struct net_eth_addr));
 }
 
+#if defined(CONFIG_NET_L2_CANBUS_ETH_TRANSLATOR)
+static bool is_for_translator(struct net_pkt *pkt)
+{
+	return ((net_pkt_lladdr_dst(pkt)->type == NET_LINK_CANBUS) &&
+		(get_lladdr(net_pkt_lladdr_dst(pkt)) ==
+		NET_CAN_ETH_TRANSLATOR_ADDR));
+}
+#endif
+
 static size_t total_lladdr_len(struct net_pkt *pkt)
 {
+#if defined(CONFIG_NET_L2_CANBUS_ETH_TRANSLATOR)
+	/* This pkt will be farowarded to Ethernet
+	 * Destination MAC is carried inline, source is going to be extended
+	 */
+	if (is_for_translator(pkt)) {
```

Comment:
What about doing it like this instead:
```
if (IS_ENABLED(CONFIG_NET_L2_CANBUS_ETH_TRANSLATOR) && is_for_translator(pkt)) {
```
Then you could also remove the #ifdef's around `is_for_translator()`